demo_modules/slither/slither.js

```diff
@@ -0,0 +1,210 @@
+const register = require('register');
+const ModuleInterface = require('lib/module_interface');
+const _ = require('underscore');
+const debug = require('debug');
+const state = require('state');
+const wallGeometry = require('wallGeometry');
+
+const COLORS = [
+  '#ff0000',
+  '#00ff00',
+  '#0000ff',
+  '#ff00ff',
+  '#cdc9c9',
+  '#ffdead',
+  '#bebebe',
+  '#e6e6fa',
+  '#8470ff',
+  '#00bfff',
+  '#40e0d0',
+  '#2e8b57',
+  '#32cd32',
+  '#ffff00',
+  '#ffd700',
+  '#cd5c5c',
+  '#b22222',
+  '#f4a460',
+  '#ffa500',
+  '#ff6347',
+  '#d02090',
+  '#8a2be2',
+]
+const SPEED = 0.2;
+
+const isOutOfBounds = ({x, y}, rect) => {
+  return x < rect.x || y < rect.y || x > rect.x + rect.w || y > rect.y + rect.h;
+};
+
+const darken = (color, i) => {
+  return `#${color.substr(1).match(/.{2}/g).map(c => Math.max(0, parseInt(c, 16) - 5*i).toString(16).padStart(2, '0')).join('')}`;
+};
+
+class SlitherServer extends ModuleInterface.Server {
+  constructor(config) {
+    super();
+    // Last position and heading and color of each snake.
+    this.snakes = [];
+    
+    this.numSnakes = config.numSnakes || 10;
+  }
+  willBeShownSoon(deadline) {
+    // Initialize the snakes.
+    this.snakes = Array.from({length: this.numSnakes}).map(() => {
+      return {
+        startTime: deadline || 0,
+        heading: Math.random() * 2 * Math.PI,
+        position: { 
+          x: Math.random() * wallGeometry.extents.w,
+          y: Math.random() * wallGeometry.extents.h,
+        },
+        color: COLORS[Math.floor(Math.random() * COLORS.length)]
+      };
+    });
+    
+    state.create('snakes', [{
+      startTime: 'ValueNearestInterpolator',
+      heading: 'NumberLerpInterpolator',
+      position: {
+        x: 'NumberLerpInterpolator',
+        y: 'NumberLerpInterpolator'
+      },
+      color: 'ValueNearestInterpolator'
+    }]);
+    return Promise.resolve();
+  }
+  tick(time, delta) {
+    if (!time) {
```

Comment:
I filed #201 to track fixing the time checks.

Reply:
Thanks!